### ISSUE TITLE
StateManager cannot persist state after huge amount of requests

### ISSUE THREAD

**Issue by e-St** [dapr/js-sdk#317]
## Expected Behavior

When I create e.g. 100.000 actors and every actor persists some state to Redis, all requests should be successful.

## Actual Behavior

After persisting the state for about 28.000 actors, I get the following error continuously:
```
FetchError: request to http://127.0.0.1:3500/v1.0/actors/MyActor/92eed1ab-7af3-4e64-a8f7-c29269e5733f/state/mystatefield failed, reason: connect EADDRNOTAVAIL 127.0.0.1:3500 - Local (127.0.0.1:0)
  at ClientRequest. (/app/node_modules/node-fetch/lib/index.js:1491:11)
  at ClientRequest.emit (node:events:527:28)
  at Socket.socketErrorListener (node:_http_client:454:9)
  at Socket.emit (node:events:527:28)
  at emitErrorNT (node:internal/streams/destroy:164:8)
  at emitErrorCloseNT (node:internal/streams/destroy:129:3)
  at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  type: 'system',
  errno: 'EADDRNOTAVAIL',
  code: 'EADDRNOTAVAIL'
  }
```

## Steps to Reproduce the Problem
I am using dapr 1.7.1 on Kubernetes
Base image for my service: node:17-alpine
js-sdk version: "@dapr/dapr": "2.3.0"

Steps to reproduce:

Define an actor class (excerpt myactor.js)

```
async init() {}

async onActivate() {
    let stateManager = this.getStateManager(this);

    await stateManager.setState("mystatefield", "a");
    await stateManager.setState("mystatefield2", "b");
    await stateManager.setState("mystatefield3", "c");
    
    await stateManager.saveState();
}

```

Within the service, create 100.000 actors in a loop (excerpt myservice.js)
```
  const amount = 100000;
  const builder = new ActorProxyBuilder(MyActor, client);

  for (let i = 0; i < amount; i++) {
    let actor = builder.build(ActorId.createRandomId());
    await actor.init()
  }
```
(Adding timeouts of 10seconds e.g. after creation of 500 actors at a time did not change anything)

My guess is, that this is a similar issue as https://github.com/nodejs/node-v0.x-archive/issues/6833
Maybe the ActorClient should create an "Agent" which is used for the http calls as discussed there?
I went through the code and node-fetch seems to be used...

Also, I tried to create 4 x 25.000 actors instead of 1 x 100.000 by spinning up 4 replicas of my service.
I did not get the error then. This makes me believe, that it has definitely something to do with node.js/the js-dapr-sdk and the way it does http requests.

**Comment by XavierGeerinck** [comment]
Tracked in #320

**Comment by shubham1172** [comment]
The fix is in and released with v2.4.0.
